### PR TITLE
GH actions updates

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

This _should_ make GH actions jobs more reliable

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -9,214 +9,276 @@ concurrency:
...
+        name: Check for build cache
         id: build-cache
         with:
           path: |
             **/dist
             !**/node_modules
           key: ${{ runner.os }}-build-cache-${{ hashFiles('**/rollup.config.js', '**/babel.config.js', '**/src/', '**/scripts/', '!**/node_modules', '!**/dist') }}
+
+      # Only setup & build if there was no build cache hit
+      - name: Use Node.js 14
```
Comment (brookescarlett) [inline .github/workflows/nodejs.yml:112]: will there ever be a cache hit at this point in the workflow? i thought github actions didn't persist caches between workflows, but i could definitely be wrong about that

Comment (brookescarlett) [reply .github/workflows/nodejs.yml:112]: nvm i think i am wrong

Comment (brookescarlett) [reply .github/workflows/nodejs.yml:112]: but i do think we need to give this cache a key?

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -9,214 +9,276 @@ concurrency:
...
-      - name: tests
-        run: yarn test --ssr --diff --deps --no-cache --ci --runInBand --reporters=default --reporters=jest-junit
-        env:
-          JEST_JUNIT_OUTPUT: 'reports/junit/js-test-ssr-results.xml'
+        if: ${{ steps.build-cache.outputs.cache-hit != 'true' }}
+        run: yarn build:packages
 
-  lints:
-    name: Check lints
+  build-ts:
```
[the remark this answers is not in the document]
Comment (bruugey) [reply .github/workflows/nodejs.yml:128]: and if something breaks in building, it'll throw right?

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (bruugey) [reply .github/workflows/nodejs.yml:93]: i thoguht `key` was required: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action

File: .github/workflows/nodejs.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply .github/workflows/nodejs.yml:109]: 👍 correct

**Review by bruugey — APPROVED** [review]

**Pull request merged**